src/ImageSharp/Formats/Tiff/Ifd/DirectoryReader.cs

```diff
@@ -14,92 +14,82 @@ namespace SixLabors.ImageSharp.Formats.Experimental.Tiff
     /// </summary>
     internal class DirectoryReader
     {
-        private readonly TiffStream stream;
-
-        private readonly EntryReader tagReader;
+        private readonly Stream stream;
 
         private uint nextIfdOffset;
 
-        public DirectoryReader(TiffStream stream)
+        // used for sequential read big values (actual for multiframe big files)
+        // todo: different tags can link to the same data (stream offset) - investigate
+        private readonly SortedList<uint, Action> lazyLoaders = new SortedList<uint, Action>(new DuplicateKeyComparer<uint>());
+
+        public DirectoryReader(Stream stream) => this.stream = stream;
+
+        public ByteOrder ByteOrder { get; private set; }
+
+        public IEnumerable<ExifProfile> Read()
         {
-            this.stream = stream;
-            this.tagReader = new EntryReader(stream);
+            this.ByteOrder = ReadByteOrder(this.stream);
+            this.nextIfdOffset = new HeaderReader(this.stream, this.ByteOrder).ReadFileHeader();
+            return this.ReadIfds();
         }
 
-        public IEnumerable<IExifValue[]> Read()
+        private static ByteOrder ReadByteOrder(Stream stream)
         {
-            if (this.ReadHeader())
+            var headerBytes = new byte[2];
```

Comment:
Stackalloc span?

Reply:
sorry, do you mean doing `new Span(stackalloc byte[2])` ?